### PR TITLE
[SYCL][ESIMD] Address failures in ESIMD Emulator e2e tests

### PR DESCRIPTION
**Pull request by turinevgeny**
* Update CM Emulator to version 1.0.31, ubu22 package to align with CI testing;
* Add libdw1 package as a new dependency of cm-cpu-emulation package.

### CODE REVIEW

**Review by v-klochkov — APPROVED** [review]

**Review by bader — CHANGES_REQUESTED** [review]
@turinevgeny, please, fix pre-commit failures.

**Review by dongkyunahn-intel — APPROVED** [review]

File: devops/scripts/install_build_tools.sh [hunk 1]
```diff
@@ -16,6 +16,7 @@ apt update && apt install -yqq \
       libffi-dev \
       libva-dev \
       libtool \
+      libdw1 \
```
Comment (bader) [inline devops/scripts/install_build_tools.sh:19]: Are all these libraries added for `cm-cpu-emulation`? I think it's worth adding a comment to this script.

**Review by bader — COMMENTED** [review]
There are two changes in this patch, which I would prefer to commit separately. Missing dependency for Ubuntu 22 is not related to version update.
I'm okay to commit them together and fix remaining issues in follow-up patches as ESIMD validation is broken, but we need to update PR title and description.

File: devops/scripts/install_build_tools.sh (hunk 1 above)
Comment (turinevgeny) [reply devops/scripts/install_build_tools.sh:19]: I do not know that. I can say that `libffi-dev` is used by `cm-cpu-emulation`, but that does not mean it is not used by something else.

**Comment by turinevgeny** [comment]
> There are two changes in this patch, which I would prefer to commit separately. Missing dependency for Ubuntu 22 is not related to version update. I'm okay to commit them together and fix remaining issues in follow-up patches as ESIMD validation is broken, but we need to update PR title and description.

Updated PR title and the description.

**Review by bader — APPROVED** [review]
Please, make two follow-up changes:
1. Update ESIMD test status on Windows to fix pre-commit.
2. Add comments about packages required by `cm-cpu-emulator`.

**Pull request merged**